it/connectors/IncomeTaxUserDataConnectorSpec.scala

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package connectors
+
+import com.github.tomakehurst.wiremock.http.HttpHeader
+import connectors.httpParsers.IncomeTaxUserDataHttpParser.IncomeTaxUserDataResponse
+import models.{APIErrorBodyModel, APIErrorModel, IncomeTaxUserData}
+import play.api.http.Status._
+import play.api.libs.json.Json
+import play.api.test.Helpers.OK
+import uk.gov.hmrc.http.HeaderCarrier
+import utils.IntegrationTest
+
+import scala.concurrent.Await
+import scala.concurrent.duration.Duration
+
+class IncomeTaxUserDataConnectorSpec extends IntegrationTest{
+
+  lazy val connector = app.injector.instanceOf[IncomeTaxUserDataConnector]
+
+  val taxYear = 2022
+
+  implicit override val headerCarrier: HeaderCarrier = HeaderCarrier().withExtraHeaders("mtditid" -> mtditid, "X-Session-ID" -> sessionId)
+
+  "IncomeTaxUserDataConnector" should {
+    "Return a success result" when {
+      "submission returns a 204" in {
+
+        stubGetWithHeadersCheck(s"/income-tax-submission-service/income-tax/nino/$nino/sources/session\\?taxYear=$taxYear", NO_CONTENT,
+        "{}",("X-Session-ID" -> sessionId), ("mtditid" -> mtditid))
+
+        val result: IncomeTaxUserDataResponse = Await.result(connector.getUserData(nino, taxYear), Duration.Inf)
+        result shouldBe Right(IncomeTaxUserData())
+      }
+
+      "submission returns a 200" in {
+
+        stubGetWithHeadersCheck(s"/income-tax-submission-service/income-tax/nino/$nino/sources/session\\?taxYear=$taxYear", OK,
+        Json.toJson(userData(employmentsModel)).toString(),("X-Session-ID" -> sessionId), ("mtditid" -> mtditid))
+
+        val result: IncomeTaxUserDataResponse = Await.result(connector.getUserData(nino, taxYear), Duration.Inf)
+        result shouldBe Right(userData(employmentsModel))
+      }
+
+      "submission returns a 200 but invalid json" in {
```

Comment:
The following test are all error cases, but are listed under "success result". Is this correct?

Reply:
Good spot, I've moved the error scenarios under their own heading